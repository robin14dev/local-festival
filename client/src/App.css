@import url(./styles/reset.css);
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

:root {
  --primaryBlue: rgb(64 82 255);
  --primaryGreen: #60ff53;
  --primaryOrange: #ff9a62;
  --primaryPurple: #6268ff;
  --mainColor: var(--primaryPurple);
  --primaryPink: #fb1039;
  --lightBlack: #000000b3;
  --rem: 16;
}

button {
  background: inherit;
  border: none;
  box-shadow: none;
  border-radius: 0;
  padding: 0;
  overflow: visible;
  cursor: pointer;
}

/* input 기본 스타일 초기화 */
input {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  outline: none;
  border: none;
}

textarea {
  border: none;
  resize: none;
}

textarea:focus,
input:focus {
  outline: none;
}

.container {
  opacity: 1;
  background-color: yellow;
  transition: all 0.3s ease-out;
}

.container.unmount {
  opacity: 0;
}

@font-face {
  font-family: 'EarlyFontDiary';
  /* src: url('https://cdn.jsdelivr.net/gh/projectnoonnu/noonfonts_220508@1.0/EarlyFontDiary.woff2') format('woff2'); */
  src: url('./fonts/EF_Diary.ttf');
  font-weight: normal;
  font-style: normal;
}

@font-face {
  font-family: 'ulsanjunggu';
  /* src: url('https://cdn.jsdelivr.net/gh/projectnoonnu/noonfonts_2206-02@1.0/ulsanjunggu.woff2') format('woff2'); */
  src: url('./fonts/ulsanjunggu.ttf');
  font-weight: normal;
  font-style: normal;
}

@font-face {
  font-family: 'HS-Regular';
  src: url('https://cdn.jsdelivr.net/gh/projectnoonnu/noonfonts_2201-2@1.0/HS-Regular.woff')
    format('woff');
  font-weight: normal;
  font-style: normal;
}

@font-face {
  font-family: 'GangwonEdu_OTFBoldA';
  src: url('https://cdn.jsdelivr.net/gh/projectnoonnu/noonfonts_2201-2@1.0/GangwonEdu_OTFBoldA.woff')
    format('woff');
  font-weight: normal;
  font-style: normal;
}

@font-face {
  font-family: 'SuseongDotum';
  src: url('https://cdn.jsdelivr.net/gh/projectnoonnu/noonfonts_2205@1.0/SuseongDotum.woff2')
    format('woff2');
  font-weight: normal;
  font-style: normal;
}
@font-face {
  font-family: 'NanumSquareRound';
  src: url('https://cdn.jsdelivr.net/gh/projectnoonnu/noonfonts_two@1.0/NanumSquareRound.woff')
    format('woff');
  font-weight: normal;
  font-style: normal;
}

@keyframes toast-in-right {
  from {
    transform: translateX(100%);
  }
  to {
    transform: translateX(0);
  }
}

@keyframes button-loading-spinner {
  from {
    transform: rotate(0turn);
  }

  to {
    transform: rotate(1turn);
  }
}

@keyframes slide-down {
  from {
    transform: translateY(-1rem);
    opacity: 0;
  }

  to {
    opacity: 1;
    transform: translateY(0rem);
  }
}

@keyframes slide-up {
  from {
    transform: translateY(0rem);
    opacity: 1;
  }

  to {
    opacity: 0;
    transform: translateY(-1rem);
  }
}

@keyframes dark-soft {
  from {
    background: none;
  }

  to {
    background: rgba(0, 0, 0, 0.4);
  }
}
@keyframes bright-soft {
  from {
    background: rgba(0, 0, 0, 0.4);
  }

  to {
    background: none;
  }
}
