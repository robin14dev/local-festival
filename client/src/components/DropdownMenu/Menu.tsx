import React from "react";
import { useNavigate } from "react-router-dom";
import { Ul } from "./styled";

export default function Menu() {
  let navigate = useNavigate();
  const onClickLogout = () => {
    sessionStorage.clear();
    navigate("/", { replace: true });
  };

  return (
    <Ul data-testid="Menu">
      <li data-testid="Menu-AccountPage" onClick={() => navigate("/Account")}>
        계정
      </li>
      <li data-testid="Menu-WishlistPage" onClick={() => navigate("/Wishlist")}>
        위시리스트
      </li>
      <li data-testid="Menu-Logout" onClick={onClickLogout}>
        로그아웃
      </li>
    </Ul>
  );
}
