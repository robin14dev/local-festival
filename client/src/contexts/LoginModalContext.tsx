import LoginModal from "components/account/LoginModal";
import React, { useState, createContext } from "react";

type State = boolean;
type Dispatch = React.Dispatch<React.SetStateAction<State>>;

export const LoginModalStateContext = createContext<State>(false);

export const LoginModalDispatchContext = createContext<Dispatch>(() => {});

type Props = {
  children: JSX.Element | JSX.Element[];
};
export const LoginModalContextProvider = ({ children }: Props) => {
  const [isLoginModal, setIsLoginModal] = useState(false);
  return (
    <LoginModalStateContext.Provider value={isLoginModal}>
      <LoginModalDispatchContext.Provider value={setIsLoginModal}>
        {isLoginModal && <LoginModal/>}
        {children}
      </LoginModalDispatchContext.Provider>
    </LoginModalStateContext.Provider>
  );
};

/**
 * ModalContext로 해서 여러개 나올 수 있게 하는 방법 
 * 
 * showModal setShowModal 해서
 * 
 * 모달 나올때 키랑 컴포넌트 맵핑해서 보여주기 
 * 
 * 배열로 넣어서 렌더링
 * 
 * 
 * 
 */
