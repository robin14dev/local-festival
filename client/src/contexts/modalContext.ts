import { createContext } from 'react';

export const ModalContext = createContext<ModalContext | null>(null);

export const induceLogin = (modalContext: ModalContext | null) => {
  modalContext && modalContext.setLoginModal(true);
};
