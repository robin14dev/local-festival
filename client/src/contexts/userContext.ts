import { createContext } from "react";

export const UserContext = createContext<UserContext>({
  authState: {
    userId: 0,
    account: "",
    nickname: "",
    defaultPic: "",
    loginStatus: false,
  },
  setAuthState: () => undefined,
});

/**
 *
 * Context 밖으로 분리하는 거 해야됨
 */
