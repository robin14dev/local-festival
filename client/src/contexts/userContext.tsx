import React, { createContext, useState } from "react";

const UserContext = createContext<UserContext>({
  authState: {
    userId: 0,
    account: "",
    nickname: "",
    defaultPic: "",
    loginStatus: false,
  },
  setAuthState: (): void => {},
});

type Props = {
  children: JSX.Element | JSX.Element[];
};

const getUserInfoFromStorage: () => AuthState = () => {
  if (sessionStorage.getItem("user")) {
    const user = sessionStorage.getItem("user");
    if (user) {
      return JSON.parse(user);
    }
  } else {
    return {
      userId: 0,
      account: "",
      nickname: "",
      defaultPic: "",
      loginStatus: false,
    };
  }
};

const UserContextProvider = ({ children }: Props) => {
  const [authState, setAuthState] = useState(getUserInfoFromStorage());

  return (
    <UserContext.Provider value={{ authState, setAuthState }}>
      {children}
    </UserContext.Provider>
  );
};

export { UserContext, UserContextProvider };
