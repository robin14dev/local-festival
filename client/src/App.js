import React, { useEffect, useState } from "react";
import Header from "./components/Header";
import Mainpage from "./pages/Mainpage";
import MyPick from "../src/pages/MyPick";
import Footer from "./components/Footer";
import { Routes, Route } from "react-router-dom";
import "./App.css";
import Detailviewpage from "./pages/Detailviewpage";
import styled from "styled-components";
import axios from "axios";
import Signup from "./components/Signup";
import AccountSetting from "./pages/AccountSetting";
const Wrapper = styled.div`
  width: 100%; //1425px 스크롤바 생김
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  margin: 0 ;
  padding: 0;
`;

function App() {
  const [authState, setAuthState] = useState({
    userId: "",
    account:"",
    nickname: "",
    loginStatus: false,
  });
  const [festivalData, setFestivalData] = useState(null);
  const [pickItems, setPickItems] = useState([]);
  const [filteredData, setFilteredData] = useState(festivalData);

  // const infiniteScroll = (addData) => {
  //   //console.log('addData', addData);
  //   // console.log('here');
  //   console.log(festivalData, addData);
  //   let addToData = festivalData.concat(addData)
  //   console.log('new state',addToData);
  //   //alert('here')
  //   setFestivalData(addToData)
  //   setFilteredData(addToData)
  // }
 
  const loginHandler = ( userId,account,nickname, loginStatus) => {
    // isLogin ? setIsLogin(false) : setIsLogin(true);
    // console.log(nickname, userId, loginStatus);
    //* 로그인한 후의 유저정보 상태변경입니다.
    const nextState = {
      userId: userId,
      account: account,
      nickname: nickname,
      loginStatus: loginStatus,
    };
    setAuthState(nextState);
    //console.log("나 실행???");

    //# 유저별 찜한 축제 가져오기
    axios.get(`${process.env.REACT_APP_SERVER_ADDRESS_LOCAL}/pick`
    ,{headers: {
      accesstoken: sessionStorage.getItem("accesstoken"),
    }}
    ).then((response) => { 
      const pickedFestivalId = response.data;
      //console.log(pickedFestivalId);
      // const festivalIdArr = pickedFestivalId.map(ele => ele.local_id)
      // const pickedFestivalByUser = festivalData.filter(ele => festivalIdArr.indexOf(ele.id) > -1)
      //{festivalId: 4} 
      setPickItems(pickedFestivalId);
    });
  };
  const onSearch = (searchText) => {

    //에러 : 경상도라고 치면 안나옴 => 경상남도, 경상북도 다 나오게
    // 6월 
    // 서울시,인천시,부산시,광주시,대전시,울산시,대구시 => '시' 빼기
    // 경상도, 전라도, 충청도 => '도' 빼기
    // * 추후 수정 : 무한스크롤 구현 이후, 렌더링 되기전의 축제들을 검색했을때도 나와야 하므로 이때는 db에서 조회해야됨.
    // * 스크롤 하면서 넘어온 정보들 안에 원하는 검색결과가 있으면 클라이언트에서 보내주고, 없으면 db에 요청

    console.log(searchText);
    // setCondition(searchText);
    const filteredFestival = festivalData.filter(
      (festival) =>
        festival.location.includes(searchText) ||
        festival.title.includes(searchText) ||
        festival.location.includes(searchText)
    );

    setFilteredData(filteredFestival);
    //alert(`${filteredFestival.length}개의 축제가 진행중입니다.`)
    // }
  };

  const resetCondition = () => {
    setFilteredData(festivalData);
  };

  const togglePick = (newPick) => {
    //#1. 픽했는지 아닌지 부터 확인
    console.log('festival', newPick);
    const found = pickItems.filter((el) => el.festivalId === newPick.festivalId);
    console.log('wht is found', found);
    if (found.length !== 0) { // 이미 찜목록에 있으면 해제를 시켜줘야됨
      console.log("found");

    //#2-1. 픽 해제해서 서버에 픽 해제한 정보 보내주기
      console.log("removeId what!!!", newPick);

      //*서버에 삭제요청 보내기
      axios.delete(`${process.env.REACT_APP_SERVER_ADDRESS_LOCAL}/pick`,  {data : {festivalId: newPick.festivalId}, headers: {
        accesstoken: sessionStorage.getItem("accesstoken"),
      }})
      .then(response => {
        console.log(response.data.message);
      })
      .catch(err => {
        console.log(err);
      })

        setPickItems(pickItems.filter((el) => el.festivalId !== newPick.festivalId));
    } else {
      console.log("add new");
     //#2-2. 픽해서 서버에 픽한 정보 보내주기
      axios
        .post(`${process.env.REACT_APP_SERVER_ADDRESS_LOCAL}/pick`, {
          festivalId: newPick.festivalId,
        }, {headers: {
          accesstoken: sessionStorage.getItem("accesstoken"),
        }})
        .then((response) => {
          console.log(response.data.message);
        })
        .catch((err) => {
          console.log(err);
        });

      setPickItems([
        ...pickItems,
        newPick
      ]);
    }
  };
 
  const handleAuthState = (nickname)=>{
    const nextAuthState = authState
    nextAuthState.nickname = nickname
    setAuthState(nextAuthState)
   
  }

  
 

  useEffect(() => {
   
    const fetchData = async () => {
      try {
        const response = await axios.get(
          `${process.env.REACT_APP_SERVER_ADDRESS_LOCAL}/festivals`,
          {params: {limit: 10}}, 
         
        );
        if (response) {
          setFestivalData(response.data);
          setFilteredData(response.data);
        } else {
          console.log("no fetch data & use dummyData");
          setFestivalData([]);
        }
        // setFestivalData(dummyData);
      } catch (error) {
        console.log(error);
        console.log(
          "error났을때 client에 저장해놓은 dummydata(20개짜리)로 렌더링합니다"
        );
        setFestivalData([]);
      }
    };
    fetchData();
    
    if(sessionStorage.getItem("accesstoken")){
      axios.get(`${process.env.REACT_APP_SERVER_ADDRESS_LOCAL}/users`, {
        headers: {
          accesstoken: sessionStorage.getItem("accesstoken"),
        },
      })
      .then(response => {
        //console.log(response.data);
        const {userId,account, nickname} = response.data.data
       
        setAuthState({
          userId :userId,
          account:account,
          nickname : nickname,
          loginStatus : true
  
        })
  
        //* 새로고침시 유저가 픽한 상태도 유지되야 하므로
        axios.get(`${process.env.REACT_APP_SERVER_ADDRESS_LOCAL}/pick`, {headers: {
        accesstoken: sessionStorage.getItem("accesstoken"),
      }}
      ).then((response) => {
        //console.log(response.data.data);
        const pickedFestival = response.data;
        //console.log(pickedFestivalId);
        // const festivalIdArr = pickedFestivalId.map(ele => ele.local_id)
        // const pickedFestivalByUser = festivalData.filter(ele => festivalIdArr.indexOf(ele.id) > -1)
        //{festivalId: 4}
        setPickItems(pickedFestival);
      });
    
        
      })
      .catch(err => {
        console.log(err);
        console.log('에러 발생');
      })
    }


    
  }, []);

  
  return (
    <Wrapper>
      <Header loginHandler={loginHandler} authState={authState} />
      <Routes>
        <Route
          exact
          path="/"
          element={
            <Mainpage
              authState={authState}
              togglePick={togglePick}
              onSearch={onSearch}
              filteredData={filteredData}
              pickItems={pickItems}
              resetCondition={resetCondition}
              // infiniteScroll={infiniteScroll}
            />
          }
        ></Route>
        <Route
          exact
          path="/MyPick"
          element={
            <MyPick
              authState={authState}
              handleAuthState={handleAuthState}
              festivalData={festivalData}
              pickItems={pickItems}
              togglePick={togglePick}
            />
          }
        ></Route>
        <Route
          exact
          path="/Detailviewpage/festivalId/:id"
          element={<Detailviewpage  
            pickItems={pickItems} 
            togglePick={togglePick} 
            authState={authState}
            />}
        ></Route>
        <Route
        exact
        path="/AccountSetting"
        element = {
          <AccountSetting handleAuthState={handleAuthState} authState={authState}/>
        }
        >
          
        </Route>
        <Route exact path="/Signup" element={<Signup />}></Route>
      </Routes>
      <Footer />
    </Wrapper>
  );
}

export default App;
