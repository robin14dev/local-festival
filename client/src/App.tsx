import "./App.css";
import "../src/styles/common.scss";
import theme from "./styles/theme";
import React, { useState, useRef, useCallback } from "react";
import { Routes, Route } from "react-router-dom";
import { Helmet } from "react-helmet";
import styled, { ThemeProvider } from "styled-components";
import axios from "axios";
import { UserProvider } from "./contexts/userContext";
import { ModalContext } from "./contexts/modalContext";
import Wishlist from "./pages/Wishlist";
import DetailView from "./pages/DetailView";
import Main from "./pages/Main";
import Account from "./pages/Account";
import Header from "./components/Header";
import Footer from "./components/Footer";
import LoginModal from "./components/account/LoginModal";

const Wrapper = styled.div`
  width: 100%;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  margin: 0;
  padding: 0;
`;

const getUserPicksFromStorage: () => FestivalItem[] = () => {
  if (sessionStorage.getItem("picks")) {
    const picks = sessionStorage.getItem("picks");
    if (picks) {
      return JSON.parse(picks);
    }
  } else {
    return [];
  }
};

function App() {
  const [festivalData, setFestivalData] = useState<FestivalItem[]>([]);
  const [pickItems, setPickItems] = useState<FestivalItem[]>(
    getUserPicksFromStorage()
  );
  const [filteredData, setFilteredData] =
    useState<FestivalItem[]>(festivalData);
  const [isLoginModal, setIsLoginModal] = useState(false);
  const offset = useRef(0);

  const togglePick: togglePick = useCallback(
    async (newPick) => {
      //#1. 픽했는지 아닌지 부터 확인
      const found = pickItems.filter(
        (el) => el.festivalId === newPick.festivalId
      );
      if (found.length !== 0) {
        // 이미 찜목록에 있으면 해제를 시켜줘야됨
        //픽 해제해서 서버에 픽 해제한 정보 보내주기
        try {
          await axios.delete(`${process.env.REACT_APP_SERVER_URL}/pick`, {
            data: { festivalId: newPick.festivalId },
            headers: {
              accesstoken: sessionStorage.getItem("accesstoken") ?? "",
            },
          });
          const nextPicks = pickItems.filter(
            (item) => item.festivalId !== newPick.festivalId
          );
          setPickItems(nextPicks);
          sessionStorage.setItem("picks", JSON.stringify(nextPicks));
        } catch (error) {
          console.log(error);
        }
      } else {
        //픽해서 서버에 픽한 정보 보내주기
        try {
          await axios.post(
            `${process.env.REACT_APP_SERVER_URL}/pick`,
            {
              festivalId: newPick.festivalId,
            },
            {
              headers: {
                accesstoken: sessionStorage.getItem("accesstoken") ?? "",
              },
            }
          );
          const nextPicks = [newPick, ...pickItems];

          setPickItems(nextPicks);
          sessionStorage.setItem("picks", JSON.stringify(nextPicks));
        } catch (error) {
          console.log(error);
        }
      }
    },
    [pickItems]
  );
  const getPickedItems = async () => {
    try {
      const result = await axios.get(
        `${process.env.REACT_APP_SERVER_URL}/pick`,
        {
          headers: {
            accesstoken: sessionStorage.getItem("accesstoken") ?? "",
          },
        }
      );

      setPickItems(result.data);
      sessionStorage.setItem("picks", JSON.stringify(result.data));
    } catch (error) {
      console.log(error);
    }
  };
  return (
    <ThemeProvider theme={theme}>
      <ModalContext.Provider value={{ isLoginModal, setIsLoginModal }}>
        <UserProvider>
          <Wrapper>
            <Helmet>
              <title>이번주엔 어디로 가볼까? - LOCO</title>
            </Helmet>
            {isLoginModal && <LoginModal setIsLoginModal={setIsLoginModal} />}

            <Header />
            <Routes>
              <Route
                path="/"
                element={
                  <Main
                    togglePick={togglePick}
                    filteredData={filteredData}
                    pickItems={pickItems}
                    setFestivalData={setFestivalData}
                    setFilteredData={setFilteredData}
                    offset={offset}
                  />
                }
              ></Route>
              <Route
                path=":search"
                element={
                  <Main
                    togglePick={togglePick}
                    filteredData={filteredData}
                    pickItems={pickItems}
                    setFestivalData={setFestivalData}
                    setFilteredData={setFilteredData}
                    offset={offset}
                  />
                }
              ></Route>

              <Route
                path="/Wishlist"
                element={
                  <Wishlist pickItems={pickItems} togglePick={togglePick} />
                }
              ></Route>
              <Route
                path="/Detail/:festivalId/*"
                element={<DetailView togglePick={togglePick} />}
              ></Route>
              <Route path="/Account" element={<Account />}></Route>
            </Routes>
            <Footer setIsLoginModal={setIsLoginModal} />
          </Wrapper>
        </UserProvider>
      </ModalContext.Provider>
    </ThemeProvider>
  );
}

export default App;
