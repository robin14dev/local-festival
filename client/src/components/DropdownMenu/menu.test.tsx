import React, { useContext } from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter, useLocation } from "react-router-dom";
import Menu from "./Menu";
import App from "../../App";
import { UserContext } from "../../contexts/userContext";
beforeEach(() => {
  const intersectionObserverMock = () => ({
    observe: () => null,
    disconnect: () => null,
  });
  window.IntersectionObserver = jest
    .fn()
    .mockImplementation(intersectionObserverMock);
});

describe("Menu 컴포넌트", () => {
  it(" Menu 컴포넌트를 성공적으로 렌더링 합니다. ", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Menu />
      </MemoryRouter>
    );
    const menu = screen.getByTestId("Menu");
    expect(menu).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("[계정] 메뉴 클릭시 계정 페이지로 이동합니다.", () => {
    const URLTestComponent = (): JSX.Element => {
      const { pathname } = useLocation();
      return <div>pathname is {pathname}</div>;
    };

    render(
      <MemoryRouter initialEntries={["/"]}>
        <URLTestComponent />
        <App />
        <Menu />
      </MemoryRouter>
    );

    const accountBtn = screen.getByTestId("Menu-AccountPage");
    const mainPage = screen.getByTestId("MainPage");
    const pathName = screen.getByText("pathname is /");

    expect(mainPage).toBeInTheDocument();
    expect(pathName).toBeInTheDocument();
    fireEvent.click(accountBtn);
    const accountPage = screen.getByTestId("AccountPage");
    expect(accountPage).toBeInTheDocument();
    expect(pathName.textContent).toBe("pathname is /Account");
  });
  it("[위시리스트] 메뉴 클릭시, 위시리스트 페이지로 이동합니다.", () => {
    const URLTestComponent = (): JSX.Element => {
      const { pathname } = useLocation();
      return <div>pathname is {pathname}</div>;
    };

    render(
      <MemoryRouter initialEntries={["/"]}>
        <URLTestComponent />
        <App />
        <Menu />
      </MemoryRouter>
    );

    const wishListBtn = screen.getByTestId("Menu-WishlistPage");
    const mainPage = screen.getByTestId("MainPage");
    const pathName = screen.getByText("pathname is /");

    expect(mainPage).toBeInTheDocument();
    expect(pathName).toBeInTheDocument();
    fireEvent.click(wishListBtn);
    const wishListPage = screen.getByTestId("WishlistPage");
    expect(wishListPage).toBeInTheDocument();
    expect(pathName.textContent).toBe("pathname is /Wishlist");
  });
  it("[로그아웃] 메뉴 클릭시 로그아웃 후, 메인페이지로 이동합니다.", () => {
    /**
     * TODO : 로그아웃이 되었는지 넣어야 할것 같음 => 이건 넣어야함 (있는 상태에서 하는거니깐)
     * - 로그인한 목업 상태 만들고
     * - 클릭한 후에 로그아웃 상태인지 확인
     * -
     *
     * TODO : 계정이랑 위시리스트 눌렀을 때 로그인 상태 아니면 이동 안하는거 추가해야 되지 않나?? => 이건 기능을 추가해야 되는거니깐 일단 보류
     *
     */
    const URLTestComponent = (): JSX.Element => {
      const { pathname } = useLocation();
      return <div>pathname is {pathname}</div>;
    };
    const UserContextTest = (): JSX.Element => {
      const {
        authState: { loginStatus },
      } = useContext(UserContext);
      return (
        <>
          <div>{loginStatus ? "로그인" : "로그아웃"}</div>
        </>
      );
    };

    const mockAuthState = {
      login: {
        loginStatus: true,
        userId: 1,
        account: "abc@abc.com",
        nickname: "abc",
        defaultPic: "",
      },
      logout: {
        loginStatus: false,
        userId: 0,
        account: "",
        nickname: "",
        defaultPic: "",
      },
    };
    const mockSetAuthState = jest.fn();

    render(
      <MemoryRouter initialEntries={["/Wishlist"]}>
        <UserContext.Provider
          value={{
            authState: mockAuthState.login,
            setAuthState: mockSetAuthState,
          }}
        >
          <UserContextTest />
          <URLTestComponent />
          <App />
          <Menu />
        </UserContext.Provider>
      </MemoryRouter>
    );

    const logoutBtn = screen.getByTestId("Menu-Logout");
    const wishListPage = screen.getByTestId("WishlistPage");
    const pathName = screen.getByText("pathname is /Wishlist");
    const loginStatus = screen.getByText("로그인");
    expect(wishListPage).toBeInTheDocument();
    expect(pathName).toBeInTheDocument();
    expect(loginStatus).toBeInTheDocument();

    fireEvent.click(logoutBtn);
    const mainPage = screen.getByTestId("MainPage");
    expect(mainPage).toBeInTheDocument();
    expect(pathName.textContent).toBe("pathname is /");
    expect(loginStatus.textContent).toBe("로그아웃");
  });
});
