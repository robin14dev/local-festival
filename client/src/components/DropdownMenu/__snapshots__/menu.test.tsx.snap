// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu 컴포넌트  Menu 컴포넌트를 성공적으로 렌더링 합니다.  1`] = `
<div>
  <ul
    class="sc-bcXHqe jMtBuy"
    data-testid="Menu"
  >
    <li
      data-testid="Menu-AccountPage"
    >
      계정
    </li>
    <li
      data-testid="Menu-WishlistPage"
    >
      위시리스트
    </li>
    <li
      data-testid="Menu-Logout"
    >
      로그아웃
    </li>
  </ul>
</div>
`;
